aptos-move/framework/aptos-stdlib/sources/data_structures/smart_table.move

```diff
@@ -257,7 +256,7 @@ module aptos_std::smart_table {
                         }
                     };
                 };
-                vector_index = 0;
+                starting_vector_index = 0;
```

Comment:
Can't we remove this @alnoki ?

Reply:
@aching if I remove that line the tests fail

The algorithm in question is a 2D loop: loop over obuckets, then loop over vector elements inside each bucket. Since it's paginated, the caller can start iteration at any arbitrary vector index inside the start bucket. Then once the bucket is exhausted, iteration must start a linear scan over the next bucket, hence resetting it to 0

Does the comment I added in beefbef197 provide sufficient clarity?

Reply:
Yeah, makes sense. Was looking at a pure diff and saw the different while vs for behavior. At least you saved one LOC by going to the for loop from the while loop =)